packages/flutter/lib/src/widgets/lazy_block.dart

```diff
@@ -23,19 +26,28 @@ abstract class LazyBlockDelegate {
 
   /// Whether [LazyBlock] should evict its cache of widgets returned by [buildItem].
   ///
-  /// When a [LazyBlock] receives a new configuration, it evicts its cache of
-  /// widgets if (1) the new configuration has a delegate with a different
-  /// runtimeType thant he old delegate, or (2) the [shouldRebuild] method of
-  /// the new delegate returns true when passed the old delgate.
+  /// When a [LazyBlock] receives a new configuration with a new delegate, it
+  /// evicts its cache of widgets if (1) the new configuration has a delegate
+  /// with a different runtimeType thant he old delegate, or (2) the
+  /// [shouldRebuild] method of the new delegate returns true when passed the
+  /// old delgate.
   ///
   /// When calling this function, [LazyBlock] will always pass an argument that
   /// matches the runtimeType of the receiver.
   bool shouldRebuild(LazyBlockDelegate oldDelegate);
 }
 
+/// Uses an [IndexedBuilder] to provide children for for [LazyBlock].
+///
+/// A LazyBlockBuilder rebuilds the children whenever the [LazyBlock] is
+/// rebuilt, similar to the behavior of [Builder].
+///
+/// See also [LazyBlockViewport].
 class LazyBlockBuilder extends LazyBlockDelegate {
-  LazyBlockBuilder(this.builder);
+  /// Creates a LazyBlockBuilder based on the given builder.
+  LazyBlockBuilder({ this.builder });
 
+  /// Returns a widget representing the item with the given index.
```

Comment:
Document the policy for determining what the last item will be.
Document the contract with respect to the index argument (can it be called in any order, can it be called for negative numbers or numbers beyond the last one that returned null, etc).